server/game/cards/02_SHD/units/MigsMayfeldTriggerman.ts

```diff
@@ -0,0 +1,29 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+
+export default class MigsMayfeldTriggerman extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '6910883839',
+            internalName: 'migs-mayfeld#triggerman'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addTriggeredAbility({
+            title: 'You may deal 2 damage to a unit or base.',
+            limit: AbilityHelper.limit.perRound(1),
+            when: {
+                onCardsDiscardedFromHand: (event, context) => true,
+            },
+            targetResolver: {
+                optional: true,
```

Comment:
We try to keep the `optional` at the ability level unless it only applies to one effect in a longer sequence (note that this will change the name of the pass button from 'choose no targets' to 'pass ability' which is relevant for which expectation you use; see comment below)

Reply:
Do you know of an example off the top where it is correct to have Optional inside targetResolver? Is Krayt Dragon correct in having it inside targetResolver? I saw in both ways, and I used Krayt Dragon as the template for Migs. In any event, moved up to the ability level.

Reply:
It would have to be a situation where one effect in an ability is optional but the others aren't. We haven't been 100% consistent about this rule in the past but we're trying to now, if you spot other abilities that don't do it this way feel free to update them